server/src/main/java/org/elasticsearch/transport/TransportRequestOptions.java

```diff
@@ -20,67 +20,27 @@
 package org.elasticsearch.transport;
 
 
-import io.crate.common.unit.TimeValue;
+import org.jetbrains.annotations.Nullable;
 
-public class TransportRequestOptions {
+import io.crate.common.unit.TimeValue;
 
-    private final TimeValue timeout;
-    private final Type type;
+public record TransportRequestOptions(@Nullable TimeValue timeout, Type type) {
 
-    private TransportRequestOptions(TimeValue timeout, Type type) {
-        this.timeout = timeout;
-        this.type = type;
-    }
+    public static final TransportRequestOptions EMPTY = new TransportRequestOptions(null, Type.REG);
 
-    public TimeValue timeout() {
-        return this.timeout;
+    public TransportRequestOptions(@Nullable long timeoutInMs) {
```

Comment:
this shouldn't be nullable right?

Reply:
Should be and was before as well - the annotation was just not there.

Reply:
but it's a `long` primitive.

Reply:
Oh right, I thought you commented on it in general.